### PR TITLE
refactor(insights): use query based insights in `<InsightMeta />`

### PR DESCRIPTION
**Pull request by thmsobrmlr**
## Problem

We'd like to move from `filters` to `query`.

## Changes

This uses a frontend-side conversion function to use `query`-nodes in the `<InsightMeta />` component and related sub-components.

I already merged in the stack on top of this. Here's a link for the changes without that: https://github.com/PostHog/posthog/pull/22738/files/a1b6d49a217128dcd1d714b7106046fe91fbbd52

## How did you test this code?

Used a test dashboard

### CODE REVIEW

File: frontend/src/lib/utils/eventUsageLogic.ts [hunk 1]
```diff
@@ -323,7 +324,6 @@ export const eventUsageLogic = kea<eventUsageLogicType>([
             dashboardId,
             lastRefreshed,
         }),
-        reportDashboardItemRefreshed: (dashboardItem: InsightModel) => ({ dashboardItem }),
```
Comment (thmsobrmlr) [inline frontend/src/lib/utils/eventUsageLogic.ts:326]: This actions doesn't have a listener i.e. it's not used.

File: frontend/src/lib/components/Cards/InsightCard/InsightMeta.tsx [hunk 2]
```diff
@@ -80,10 +80,7 @@ export function InsightMeta({
     const otherDashboards = nameSortedDashboards.filter((d) => !dashboards?.includes(d.id))
     const editable = insight.effective_privilege_level >= DashboardPrivilegeLevel.CanEdit
 
-    // not all interactions are currently implemented for queries
-    const allInteractionsAllowed = !insight.query
```
Comment (thmsobrmlr) [inline frontend/src/lib/components/Cards/InsightCard/InsightMeta.tsx:84]: Couldn't replicate these issues.

File: frontend/src/lib/components/Cards/InsightCard/InsightCard.tsx [hunk 3]
```diff
@@ -266,11 +266,12 @@ function InsightCardInternal(
     }: InsightCardProps,
     ref: React.Ref<HTMLDivElement>
 ): JSX.Element {
+    const insight = getQueryBasedInsightModel(legacyInsight)
```
Comment (thmsobrmlr) [inline frontend/src/lib/components/Cards/InsightCard/InsightCard.tsx:269]: We replace the insight's `filters` with a `query`, if that hasn't happened backend-side yet.

File: frontend/src/lib/components/Cards/InsightCard/InsightCard.tsx [hunk 4]
```diff
@@ -266,11 +266,12 @@ function InsightCardInternal(
...
     const { theme } = useValues(themeLogic)
     const insightLogicProps: InsightLogicProps = {
         dashboardItemId: insight.short_id,
         dashboardId: dashboardId,
-        cachedInsight: insight,
+        cachedInsight: legacyInsight, // TODO: use query based insight here
```
Comment (thmsobrmlr) [inline frontend/src/lib/components/Cards/InsightCard/InsightCard.tsx:274]: Want to do this in a separate PR, since many dragons are/might be hidden here.

File: frontend/src/lib/components/Cards/InsightCard/InsightMeta.tsx [hunk 5]
```diff
@@ -122,10 +121,10 @@ export function InsightMeta({
                     {loading && <LemonTableLoader loading={true} />}
                 </>
             }
-            metaDetails={<InsightDetails insight={insight} />}
+            metaDetails={<InsightDetails insight={legacyInsight} />}
```
Comment (thmsobrmlr) [inline frontend/src/lib/components/Cards/InsightCard/InsightMeta.tsx:124]: The `<InsightDetails />` component will be refactored in a separate PR as well. It's quite a bit of code and there are bugs present that I'd like to fix while having a look at it.

**Comment by thmsobrmlr** [comment]
Already merged in the stack on top of this. Here's a link for the changes without that: https://github.com/PostHog/posthog/pull/22738/files/a1b6d49a217128dcd1d714b7106046fe91fbbd52

**Review by webjunkie — APPROVED** [review]
